src/Symfony/Component/Cache/Adapter/FilesystemAdapter.php

```diff
@@ -138,8 +136,9 @@ protected function doSave(array $values, $lifetime)
                 @mkdir($dir, 0777, true);
             }
             $value = $expiresAt."\n".rawurlencode($id)."\n".serialize($value);
-            if (false !== @file_put_contents($file, $value, LOCK_EX)) {
-                @touch($file, $expiresAt);
+            if (false !== @file_put_contents($tmp, $value)) {
+                @touch($tmp, $expiresAt);
+                $ok = @rename($tmp, $file) && $ok;
```

Comment:
I feel like If I add a comment for this kind of code lines, I should add one on every single line... There are several places in the component (and also in doctrine cache) where this style is in use...